### DISCUSSION TITLE
Add currency code to Greenfield API Invoice Payment Methods?

### DISCUSSION THREAD

**Discussion by woutersamaey** [btcpayserver/btcpayserver#2889] [Technical Support] [answered]
When using the Greenfield API, you can get the invoice payment methods. These codes consist of a crypto code and method (OnChain / LightningNetwork).

There is however, no ISO compatible currency code, unless the crypto code is that?

If would be nice to have an ISO compatible currency code per payment, so we could build a message like the following in an application:
```
The customer paid 100 USD using 0.0023 XXX with an average value of 0.000023 USD per XXX
```
USD is the fiat currency used on the invoice
XXX would be the crypto used (BTC / ETH / XMR / ...). Having this as ISO compatible would help I believe. So BTC would in fact be XBT like Kraken does it.

Wether to use ISO or other standards is up for debate.

I'm not sure if this is what a `crypto_code` is or of there is another standard we can add to the API response as a single field, not requiring us to split the Payment Method string by `-` and using the first part.

**Comment by Kukks** [comment, marked as answer]
crypto_code is exactly that

**Comment by mp21-org** [comment]
I would like to reopen the discussion:

An Invoice/Store has several available PaymentMethods assigned to it when it was created.
However how do I know which Crypto was used for a specific Payment of the payments[] array in [GetInvoicePaymentMethods](https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Invoices_GetInvoicePaymentMethods)?

At least add crypto_code to the Payment Object.

My invoice can be partially paid with BTC and partially paid with LTC for that matter. I want to show transaction URLs(mempool.space, blockchair.com/ltc/) depending on the crypto used for each specific payment.

Hope my problem came across. Thanks.

**Reply by woutersamaey** [reply]
You'll have 2 payment methods for BTC and LTC and in each of those you will have a payments array with 1 payment in each (1 for BTC and 1 for LTC).

Does this help?

**Reply by mp21-org** [reply]
I see and the returned PaymentMethod array contains a string "paymentMethod" which would be something like "BTC, BTC-LightningNetwork" accordingly. Allowing me to handle the subsequent payments accordingly.

**Reply by woutersamaey** [reply]
Yep.
It's also a bit messy if you want to add up amounts paid for example. Payments can have different statuses, so they're not always confirmed.

**Reply by woutersamaey** [reply]
I'm open to putting some of the code in the PHP lib if it helps. We don't want all PHP devs coding the same loop if it can be avoided.
